test/src/tests-store.ts

```diff
@@ -254,6 +254,422 @@ describe('Store', () => {
+            expect(store.persistenceData[testMPID].con.gdpr).to.deep.equal(
+                expectedConsentState.gdpr
+            );
+
+            expect(store.persistenceData[testMPID].con.ccpa).be.ok;
+            expect(store.persistenceData[testMPID].con.ccpa).to.deep.equal(
+                expectedConsentState.ccpa
+            );
+        });
+
+        it('should set consent state as a minified object in persistence', () => {
+            const store: IStore = new Store(
+                sampleConfig,
+                window.mParticle.getInstance()
+            );
+
+            const consentState = window.mParticle.Consent.createConsentState();
+
+            const gdprConsent = window.mParticle
+                .getInstance()
+                .Consent.createGDPRConsent(
+                    true,
+                    10,
+                    'foo gdpr document',
+                    'foo gdpr location',
+                    'foo gdpr hardware id'
+                );
+
+            const ccpaConsent = window.mParticle
+                .getInstance()
+                .Consent.createCCPAConsent(
+                    false,
+                    42,
+                    'foo ccpa document',
+                    'foo ccpa location',
+                    'foo ccpa hardware id'
+                );
+
+            const expectedConsentState = {
+                gdpr: {
+                    analytics: {
+                        c: true,
+                        d: 'foo gdpr document',
+                        h: 'foo gdpr hardware id',
+                        l: 'foo gdpr location',
+                        ts: 10,
+                    },
+                },
+                ccpa: {
+                    data_sale_opt_out: {
+                        c: false,
+                        d: 'foo ccpa document',
+                        h: 'foo ccpa hardware id',
+                        l: 'foo ccpa location',
+                        ts: 42,
+                    },
+                },
+            };
+
+            consentState.addGDPRConsentState('analytics', gdprConsent);
+            consentState.setCCPAConsentState(ccpaConsent);
+
+            store.setConsentState(testMPID, consentState);
+
+            const fromPersistence = window.mParticle
+                .getInstance()
+                ._Persistence.getPersistence();
+
+            expect(fromPersistence[testMPID]).to.be.ok;
+            expect(fromPersistence[testMPID].con).to.be.ok;
+
+            expect(fromPersistence[testMPID].con.gdpr).to.be.ok;
+            expect(fromPersistence[testMPID].con.gdpr).to.deep.equal(
+                expectedConsentState.gdpr
+            );
+
+            expect(fromPersistence[testMPID].con.ccpa).to.be.ok;
+            expect(fromPersistence[testMPID].con.ccpa).to.deep.equal(
+                expectedConsentState.ccpa
+            );
+        });
+
+        it('should override persistence with store values', () => {
```

Comment:
OK, added a two step process to add an additional GDPR purpose. Let me know if that works for you. I'm intentionally being verbose so that it's more obvious that we're adding extra consent purposes that are outside of the norm.